File: core/src/main/java/com/egoal/darkestpixeldungeon/actors/buffs/Shock.java
package com.egoal.darkestpixeldungeon.actors.buffs;

import com.egoal.darkestpixeldungeon.messages.Messages;
import com.egoal.darkestpixeldungeon.ui.BuffIndicator;

/**
 * Created by 93942 on 10/20/2018.
 */

// see Char::checkHit
public class Shock extends FlavourBuff{

  {
    type = buffType.NEGATIVE;
  }
  
  @Override
  public int icon() {
    return BuffIndicator.SHOCK;
  }

  @Override
  public String toString() {
    return Messages.get(this, "name");
  }

  @Override
  public String desc() {
    return Messages.get(this, "desc", dispTurns());
  }
}

File: core/src/main/java/com/egoal/darkestpixeldungeon/actors/buffs/Shock.kt
package com.egoal.darkestpixeldungeon.actors.buffs

import com.egoal.darkestpixeldungeon.messages.Messages
import com.egoal.darkestpixeldungeon.ui.BuffIndicator

/**
 * Created by 93942 on 10/20/2018.
 */

// see Char::checkHit
class Shock : FlavourBuff() {

    init {
        type = Buff.buffType.NEGATIVE
    }

    override fun icon(): Int {
        return BuffIndicator.SHOCK
    }

    override fun toString(): String {
        return Messages.get(this, "name")
    }

    override fun desc(): String {
        return Messages.get(this, "desc", dispTurns())
    }
}
